learn-array-methods-by-building-a-statistics-calculator/script.js

```diff
@@ -0,0 +1,42 @@
+const getMean = (array) => {
+  // teach the reduce method
+  const mean = array.reduce((acc, el) => acc + el, 0) / array.length;
+  return mean;
+};
+
+const getMedian = (array) => {
+  const median = array[Math.floor((array.length - 1) / 2)];
+  return median;
+};
+
+const getMode = (array) => {
+  const counts = {};
+  // teach the forEach method??
```

Comment:
I think it is fine to teach it.
The `forEach` method might have come up earlier, but this project is small enough were we can teach it/review it here too.

Reply:
I was also considering chaining the `map` and `filter` and explaining it there.

Reply:
Agreed. This would be a great place to review the `forEach` method. Currently it's being introduced in the Music Playlist project.

It would be good to review things like how the `forEach` method is great when you want to call a function for each element in an array, not when you want to return something like with `map` or `reduce`, and that `forEach` returns `undefined`.